engine/api/src/main/scala/pl/touk/nussknacker/engine/api/typed/typing.scala

```diff
@@ -129,6 +129,9 @@ object typing {
       throw new IllegalArgumentException("Cannot have typed class of Any, use Unknown")
     } else if (klass.isPrimitive) {
       TypedClass(ClassUtils.primitiveToWrapper(klass), parameters)
+    } else if (klass.isArray) {
+      //to not have separate class for each array, we pass Array of Objects
+      TypedClass(classOf[Array[Object]], List(TypedClass(ClassUtils.primitiveToWrapper(klass.getComponentType), List())))
```

Comment:
Typed.typedClass(klass.getComponentType) => to handle nested arrays ;)

Reply:
changed it